### ISSUE TITLE
Auto fix mode results in stack trace for invalid playbook

### ISSUE THREAD

**Issue by ganeshrn** [ansible/ansible-lint#3840]
##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
% ansible-lint --version
ansible-lint 6.20.3 using ansible-core:2.15.1 ansible-compat:4.1.10 ruamel-yaml:0.17.26 ruamel-yaml-clib:0.2.7
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```console (paste below)
---
- name: Test
  hosts: localhost
  gather_facts: false

  tasks:
    - name: Print hello message
      ansible.builtin.debug:
            msg: "Hello!"
          register: vm_output
```
Run ansible-lint for above playbook
```
ansible-lint playbooks/test.yml --fix all
```

##### Desired Behavior

Should report syntax-error rule violation without stacktrace
##### Actual Behavior

Please give some details of what is happening.
Include a [minimum complete verifiable example] with:

- minimized playbook to reproduce the error
- the output of running ansible-lint including the command line used
- if you're getting a stack trace, also the output of
  `ansible-playbook --syntax-check playbook`
```
% ansible-playbook --syntax-check playbooks/test.yml 
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  did not find expected key

The error appears to be in '/Users/test/Documents/ansible-lightspeed-bot/sample/lightspeed_demos_private/playbooks/test.yml': line 10, column 11, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

            msg: "Hello!"
          register: vm_output
          ^ here
```

 Stacktrace
```paste below

  File "/Users/test/Documents/venv/plugin/lib/python3.10/site-packages/ruamel/yaml/parser.py", line 585, in parse_block_mapping_key
    raise ParserError(
ruamel.yaml.parser.ParserError: while parsing a block mapping
  in "<unicode string>", line 7, column 7:
        - name: Print hello message
          ^ (line: 7)
expected <block end>, but found '<block mapping start>'
  in "<unicode string>", line 10, column 11:
              register: vm_output
              ^ (line: 10)
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
